src/item_factory.cpp

```diff
@@ -2058,6 +2056,24 @@ void Item_factory::load_basic_info( JsonObject &jo, itype &def, const std::strin
     assign( jo, "explode_in_fire", def.explode_in_fire );
     assign( jo, "insulation", def.insulation_factor );
 
+    if( jo.has_int( "weight" ) ) {
+        assign( jo, "weight", def.weight, strict, 0_gram );
+    } else if( jo.has_string( "weight" ) ) {
+        def.weight = read_from_json_string<units::mass>
+                     ( *jo.get_raw( "weight" ), units::mass_units );
+    } else {
+        def.weight = 0_gram;
```

Comment:
Doesn't this reset the weight of all items defined via "copy-from" to 0 (if the new definition does not have a separate weight entry)?